test/performance/benchmarks/dataplane-probe/main.go

```diff
@@ -0,0 +1,188 @@
+			},
+			// hitting a Knative Service
+			// going through JUST the queue-proxy falls in the +10ms range.
+			slaMin: minDefault,
+			slaMax: 110 * time.Millisecond,
+		},
+		"activator": {
+			target: vegeta.Target{
+				Method: http.MethodGet,
+				URL:    "http://activator.default.svc.cluster.local?sleep=100",
+			},
+			// hitting a Knative Service
+			// going through BOTH the activator and queue-proxy falls in the +10ms range.
+			slaMin: minDefault,
+			slaMax: 110 * time.Millisecond,
+		},
+	}
+)
+
+func main() {
+	ctx := signals.NewContext()
+	cfg := injection.ParseAndGetRESTConfigOrDie()
+	ctx, startInformers := injection.EnableInjectionOrDie(ctx, cfg)
+	startInformers()
+
+	if *target == "" {
+		log.Fatalf("-target is a required flag.")
+	}
+
+	log.Println("Starting dataplane probe for target:", *target)
+
+	ctx, cancel := context.WithTimeout(ctx, *duration+time.Minute)
+	defer cancel()
+
+	// Based on the "target" flag, load up our target benchmark.
+	// We only run one variation per run to avoid the runs being noisy neighbors,
+	// which in early iterations of the benchmark resulted in latency bleeding
+	// across the different workload types.
+	t, ok := targets[*target]
+	if !ok {
+		log.Fatalf("Unrecognized target: %s", *target)
+	}
+
+	// Make sure the target is ready before sending the large amount of requests.
+	if err := performance.ProbeTargetTillReady(t.target.URL, *duration); err != nil {
+		log.Fatalf("Failed to get target ready for attacking: %v", err)
+	}
+
+	// Send 1000 QPS (1 per ms) for the given duration with a 30s request timeout.
+	rate := vegeta.Rate{Freq: 1, Per: time.Millisecond}
+	targeter := vegeta.NewStaticTargeter(t.target)
+	attacker := vegeta.NewAttacker(vegeta.Timeout(30 * time.Second))
+
+	influxReporter, err := performance.NewInfluxReporter(map[string]string{"target": *target})
+	if err != nil {
+		log.Fatalf("failed to create influx reporter: %v", err.Error())
+	}
+	defer influxReporter.FlushAndShutdown()
+
+	// Start the attack!
+	results := attacker.Attack(targeter, rate, *duration, "load-test")
+	deploymentStatus := performance.FetchDeploymentStatus(ctx, system.Namespace(), "activator", time.Second)
+
+	var metricResults vegeta.Metrics
+
+LOOP:
+	for {
+		select {
+		case <-ctx.Done():
+			// If we time out or the pod gets shutdown via SIGTERM then start to
+			// clean thing up.
+			break LOOP
+
+		case ds := <-deploymentStatus:
+			// Report number of ready activators.
+			influxReporter.AddDataPoint(benchmarkName, map[string]interface{}{"activator-pod-count": float64(ds.ReadyReplicas)})
+
+		case res, ok := <-results:
+			if ok {
+				metricResults.Add(res)
+			} else {
+				// If there are no more results, then we're done!
+				break LOOP
+			}
+		}
+	}
+
+	// Compute latency percentiles
+	metricResults.Close()
+
+	// Report results to influx
+	influxReporter.AddDataPoint(benchmarkName, map[string]interface{}{"requests": float64(metricResults.Requests)})
```

Comment:
It seems that the stdout reporter prints mode data like throughtput and rate.
Do we want to add that to influxdb? Also do we want an SLA for request rate vs [throughput](https://github.com/knative/serving/blob/main/vendor/github.com/tsenart/vegeta/v12/lib/metrics.go#L36)?

Reply:
I think we could, yes. But in reality, if the test works as expected, the throughput + rate are static (as we define the rate that vegeta should use). Also the dashboard is already a bit crowded.
But I think it might not hurt to record those things, even if we don't visualise them. I'm going to add a function to `influxReporter` to report `metricsResults` generically.